elkm1_lib/util.py

```diff
@@ -24,6 +25,26 @@ def url_scheme_is_secure(url: str) -> bool:
     return scheme.startswith("elks")
 
 
+@cache
+def ssl_context_for_scheme(scheme: str) -> ssl.SSLContext:
+    """Create an SSL context for the given scheme.
+
+    Since ssl context is expensive to create, cache it
+    for future use since we only have a few schemes.
+    """
+    ssl_context = ssl.SSLContext(ssl.PROTOCOL_TLS)
+    if tls := TLS_VERSIONS.get(scheme):
+        ssl_context.minimum_version = tls
+        ssl_context.maximum_version = tls
+
+    ssl_context.set_ciphers(
+        "DEFAULT:!aNULL:!eNULL:!MD5:!3DES:!DES:!RC4:!IDEA:!SEED:!aDSS:!SRP:!PSK"
+    )
+    # ssl.OP_LEGACY_SERVER_CONNECT is only available in Python 3.12a4+
+    ssl_context.options |= getattr(ssl, "OP_LEGACY_SERVER_CONNECT", 0x4)
```

Comment:
We are stuck with a getattr here since https://github.com/python/cpython/issues/89051 the choice was made not to backport the new constant